### PR TITLE
store, daemon, client, cmd/snap: implement `snap find --private`

### PR DESCRIPTION
**Pull request by chipaca**
This builds on #1554 so [compare with that](https://github.com/chipaca/snappy/compare/find-search...find-private).

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -597,14 +597,32 @@ func (s *Store) Snap(name, channel string, devmode bool, user *auth.UserState) (
...
+	if private && user == nil {
+		return nil, ErrUnauthenticated
+	}
+
 	// see https://github.com/snapcore/snapd/blob/master/docs/rest.md#v2find
 
 	searchTerm = strings.TrimSpace(searchTerm)
 
 	prefix := "name"
+	if private {
```
Comment (mvo5) [inline store/store.go]: So private implies a `text` query? Or am I reading this incrrectly?

Comment (mvo5) [inline store/store.go]: Maybe a comment why this is needed? Feature? Bug?

**Comment by mvo5** [comment]
Looks fine! Maybe a small comment about private and text in the code?

**Comment by mvo5** [comment]
retest this please

File: store/store.go (hunk 1 above)
Comment (chipaca) [inline store/store.go]: that's what the store implements; see http://search.apps.ubuntu.com/docs/ (search for `private=`)

**Comment by pedronis** [comment]
lgtm

**Comment by mvo5** [comment]
:+1:

**Pull request merged**